### PR TITLE
Update go dependencies

### PR DESCRIPTION
**Pull request by andytinkham**
Signed-off-by: Andy Tinkham <andy.tinkham@cyberark.com>

### Desired Outcome

Remove dependencies on yaml.v3 prior to 3.0.1

### Implemented Changes

Updates every dependency we can currently update. Still need to update otel when an updated version becomes available.

### Connected Issue/Story
CyberArk internal issue: CONJSE-1378

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [X] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [X] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [X] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [X] No behavior was changed with this PR

#### Security

- [X] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [ ] There are no security aspects to these changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
   [cyberark/conjur-authn-k8s-client#471](https://github.com/cyberark/conjur-authn-k8s-client/pull/471)
 
 ## [0.24.4] - 2022-05-31
+### Changed
```
Comment (rpothier) [inline CHANGELOG.md:15]: Is this the correct version? Should it be 0.24.4 or 0.23.4?

Comment (andytinkham) [reply CHANGELOG.md:15]: Should be 0.23.4. Thanks for catching that.

Comment (andytinkham) [reply CHANGELOG.md:10]: Rather than complicate things, I put my change 0.23.5 to match yours.

**Review by rpothier — APPROVED** [review]
LGTM!

**Pull request merged**